src/main/java/net/syscon/elite/service/BookingService.java

```diff
@@ -669,21 +673,32 @@ public MilitaryRecords getMilitaryRecords(final Long bookingId) {
     }
 
     @Transactional
+    @VerifyBookingAccess
+    @HasWriteScope
     public void updateLivingUnit(final Long bookingId, final Long livingUnitId) {
         var offenderBooking = offenderBookingRepository.findById(bookingId)
                 .orElseThrow(EntityNotFoundException.withMessage(format("Offender booking for booking id %d not found", bookingId)));
         final var location = agencyInternalLocationRepository.findById(livingUnitId)
                 .orElseThrow(EntityNotFoundException.withMessage(format("Living unit with id %d not found", livingUnitId)));
 
+        validateUpdateLivingUnit(offenderBooking, location);
+
+        offenderBooking.setAssignedLivingUnitId(livingUnitId);
+        offenderBookingRepository.save(offenderBooking);
+        log.info("Updated offender {} booking id {} to living unit id {}", offenderBooking.getOffender().getNomsId(), offenderBooking.getBookingId(), livingUnitId);
+    }
+
+    private void validateUpdateLivingUnit(final OffenderBooking offenderBooking, final AgencyInternalLocation location) {
         checkArgument(
                 offenderBooking.getLocation().getId().equals(location.getAgencyId()),
                 "Move to living unit in prison %s invalid for offender %s in prison %s",
                 location.getAgencyId(), offenderBooking.getOffender().getNomsId(), offenderBooking.getLocation().getId()
         );
-
-        offenderBooking.setLivingUnitId(livingUnitId);
-        offenderBookingRepository.save(offenderBooking);
-        log.info("Updated offender {} booking id {} to living unit id {}", offenderBooking.getOffender().getNomsId(), offenderBooking.getBookingId(), livingUnitId);
+        checkArgument(
+                location.getLocationType().equals("CELL"),
```

Comment:
I'd consider pushing this sort of logic in the location / agency object perhaps.